File: python/silver/25206.py
def change_score(rating):
    if rating == 'A+':
        return 4.5
    elif rating == 'A0':
        return 4.0
    elif rating == 'B+':
        return 3.5
    elif rating == 'B0':
        return 3.0
    elif rating == 'C+':
        return 2.5
    elif rating == 'C0':
        return 2.0
    elif rating == 'D+':
        return 1.5
    elif rating == 'D0':
        return 1.0
    else:
        return 0.0
        

credits = 0
result = 0
for _ in range(20):
    subject, credit, rating = input().split()
    if(rating != 'P'):
        credits += float(credit)
        result += change_score(rating) * float(credit)
print(result/credits)

File: C++/silver/25206.cpp
#include <iostream>

using namespace std;

float change_score(string rating){
    float score = 0;
    switch(rating[0]){
        case 'A':
            if (rating[1] == '+') score = 4.5;
            else if (rating[1] == '0') score = 4.0;
            break;
        case 'B':
            if (rating[1] == '+') score = 3.5;
            else if (rating[1] == '0') score = 3.0;
            break;
        case 'C':
            if (rating[1] == '+') score = 2.5;
            else if (rating[1] == '0') score = 2.0;
            break;
        case 'D':
            if (rating[1] == '+') score = 1.5;
            else if (rating[1] == '0') score = 1.0;
            break;
        default:
            break;
    }
    return score;
}


int main(){
    string subject, rating;
    float credit, result, temp, credits = 0.0;
    for(int i =0 ; i< 20; i++){
        cin >> subject >> credit >> rating;
        temp = change_score(rating);
        if(rating[0] != 'P'){
            result += (temp * credit);
            credits += credit; 
        }
    }
    result /= credits;
    cout << result;
    return 0;
}